src/DynamoRevit/DynamoRevit.cs

```diff
@@ -112,42 +113,85 @@ public DynamoRevitCommandData()
-        private static bool shouldShowUi = true;
-        private static bool isAutomationMode;
-
         /// <summary>
         /// The journal file can use this key to specify whether
         /// the Dynamo UI should be visible at run time.
         /// </summary>
-        private const string JournalShowUiKey = "dynShowUI";
+        public const string JournalShowUiKey = "dynShowUI";
 
         /// <summary>
         /// If the journal file specifies automation mode, 
         /// Dynamo will run on the main thread without the idle loop.
         /// </summary>
-        private const string JounralAutomationModeKey = "dynAutomation";
+        public const string JournalAutomationModeKey = "dynAutomation";
+
+        /// <summary>
+        /// The journal file can specify a Dynamo workspace to be opened 
+        /// (and executed if we are in automation mode) at run time.
+        /// </summary>
+        public const string JournalDynPathKey = "dynPath";
 
         /// <summary>
-        /// The journal file can specify a Dynamo workspace to 
-        /// be opened (and executed) at run time.
+        /// The journal file can specify if the Dynamo workspace opened 
+        /// from JournalDynPathKey will be executed or not. 
+        /// If we are in automation mode the workspace will be executed regardless of this key.
         /// </summary>
-        private const string JournalDynPathKey = "dynPath";
+        public const string JournalDynPathExecuteKey = "dynPathExecute";
+
+        /// <summary>
+        /// The journal file can specify if the Dynamo workspace opened
+        /// from JournalDynPathKey will be forced in manual mode.
+        /// </summary>
+        public const string JournalForceManualRun = "dynForceManualRun";
+    }
+
+
+    [Transaction(TransactionMode.Manual),
+     Regeneration(RegenerationOption.Manual)]
+    public class DynamoRevit : IExternalCommand
+    {
+        enum Versions { ShapeManager = 222 };
+
+        /// <summary>
+        /// Based on the RevitDynamoModelState a dependent component can take certain 
+        /// decisions regarding its UI and functionality.
+        /// In order to be able to run a specified graph , revitDynamoModel needs to be 
+        /// at least in StartedUIless state. 
+        /// </summary>
+        public enum RevitDynamoModelState { NotStarted, StartedUIless, StartedUI };
+
+        private static List<Action> idleActions;
+        private static DynamoRevitCommandData extCommandData;
+        private static DynamoViewModel dynamoViewModel;
+        private static RevitDynamoModel revitDynamoModel;
+        private static bool handledCrash;
+        private static List<Exception> preLoadExceptions;
+
+        /// <summary>
+        /// The modelState tels us if the RevitDynamoModel was started and if has the
+        /// the Dynamo UI attached to it or not 
+        /// </summary>
+        public static RevitDynamoModelState ModelState
```

Comment:
Model state takes into account both revitDynamoModel and dynamoViewModel so for now I would say to keep it outside these two. We can assume that if dynamoViewModel is not initialized we are UI-less mode but in order to be in UI mode it looks that there are some other operations that need to be performed on dynamoViewModel . So we can't rely only on the dynamoViewModel initialization to conclude that we are in UI-less mode.